packages/web/hooks/use-performance-monitoring.ts

```
@@ -6,320 +1,114 @@
-        message: `High memory usage: ${dashboard.memoryUsage.memoryUsagePercent.toFixed(1)}%`,
-        timestamp: Date.now(),
-      })
+    
+    // Report metrics periodically
+    const reportInterval = setInterval(() => {
+      if (metricsRef.current.length > 0) {
+        reportMetrics(metricsRef.current)
+        metricsRef.current = []
+        lastReportRef.current = Date.now()
+      }
+    }, reportInterval)
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)